### PR TITLE
created new slice for Logged User

### PR DESCRIPTION
**Pull request by osaulenkoUA**
Created a new slice for logged user
All user info, for buyer or seller, is now in userState slice.

### CODE REVIEW

File: src/components/shared/Header/UserBar/index.tsx [hunk 1]
```diff
@@ -44,14 +46,14 @@ const UserPanel: FC = () => {
       </Link>
       <SearchBar />
       <span className="flex gap-6 items-center cursor-pointer">
-        {isLogged && name && lastName && (
+        {isLogged && user && user.name && user.lastName && (
```
Comment (Andrii-Pinchuk) [inline src/components/shared/Header/UserBar/index.tsx:49]: If a user is successfully logged in, he needs to receive all info about him immediately after receiving access and refresh tokens. So if the user isLogged a user should be defined.

P.S. You have UserState with a user, where all fields can be undefined and set with "?". But if a user exists he needs to have some required fields. And for now, he doesn't have any unique field (like ID).

File: src/components/shared/ModalRegistration/AuthForm/index.tsx [hunk 2]
```diff
@@ -28,15 +29,16 @@ const AuthForm: FC<AuthProps> = ({ isModalOpen, setIsModalOpen }) => {
 
     dispatch(thunkAuthSignin({ email: login, password })).then(res => {
```
Comment (Andrii-Pinchuk) [inline src/components/shared/ModalRegistration/AuthForm/index.tsx:30]: image
image
image

File: src/components/shared/ModalRegistration/AuthForm/index.tsx [hunk 3]
```diff
@@ -28,15 +29,16 @@ const AuthForm: FC<AuthProps> = ({ isModalOpen, setIsModalOpen }) => {
...
       if (res.payload.buyer) {
-        dispatch(setLastName(res.payload.buyer.lastName));
-        dispatch(setName(res.payload.buyer.name));
+        dispatch(SET_LOGGED_USER(res.payload.buyer));
+      }
+      // Для Продавця
+      if (res.payload.seller) {
+        dispatch(SET_LOGGED_USER(res.payload.seller));
       }
-      /*дописать для продавца*/
```
Comment (Andrii-Pinchuk) [inline src/components/shared/ModalRegistration/AuthForm/index.tsx:35]: Write comments in English, please. We can see payload.buyer and payload.seller. Why we need comments for this? I think is better to remove first and second comments. About the last one comment - use keywords to show that you need to do something with that in future //TODO or //todo would be great and if you have IDEA (Webstorm etc.) you can see info about todo when you create commit with this file next time.

File: src/components/shared/ModalRegistration/AuthForm/index.tsx [hunk 4]
```diff
@@ -28,15 +29,16 @@ const AuthForm: FC<AuthProps> = ({ isModalOpen, setIsModalOpen }) => {
...
     });
-
     navigate('/');
-
     setIsModalOpen(!isModalOpen);
     /*положить токен в лс */
```
Comment (Andrii-Pinchuk) [inline src/components/shared/ModalRegistration/AuthForm/index.tsx:43]: use // TODO or // todo and rewrite comment message in English, please.

File: src/store/reducers/registrationCommon.ts [hunk 5]
```diff
@@ -41,16 +41,17 @@ export const thunkAuthSignin = createAsyncThunk(
...
 
         throw new Error(`${err.message}`);
       }
       return await res.json();
     } catch (error) {
-      console.log('errror', error);
+      console.log('error', error);
 
       return rejectWithValue(getErrorMessage(error));
     }
```
Comment (Andrii-Pinchuk) [inline src/store/reducers/registrationCommon.ts:57]: You have async await syntax with try catch. Catch is needed for catching every errors
Why we need trhrow new here?
https://javascript.info/async-await
image

File: src/store/reducers/userSlice.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export class UserState {
+  user?: {
```
Comment (Andrii-Pinchuk) [inline src/store/reducers/userSlice.ts:4]: User don't have any required fields?

**Review by Andrii-Pinchuk — CHANGES_REQUESTED** [review]

File: src/components/shared/ModalRegistration/AuthForm/index.tsx (hunk 4 above)
Comment (osaulenkoUA) [reply src/components/shared/ModalRegistration/AuthForm/index.tsx:43]: /*положить токен в лс */ - who wrote? also not English, agree my fault, so TODO and in English or just commetns

File: src/store/reducers/registrationCommon.ts (hunk 5 above)
Comment (osaulenkoUA) [reply src/store/reducers/registrationCommon.ts:57]: Agree, but it's not my code. we don't need to throw new error or even alert I would say

File: src/components/shared/ModalRegistration/AuthForm/index.tsx (hunk 3 above)
Comment (osaulenkoUA) [reply src/components/shared/ModalRegistration/AuthForm/index.tsx:35]: agree my fault, I saw not in English and without any mind written in Ukrainian also

File: src/components/shared/Header/UserBar/index.tsx (hunk 1 above)
Comment (osaulenkoUA) [reply src/components/shared/Header/UserBar/index.tsx:49]: yes, we can remove ?  from user?    and then remove one more condition

File: src/components/shared/Header/UserBar/index.tsx [hunk 7]
```diff
@@ -11,15 +10,18 @@ import DropDownMenu from '../../../common/DropDownMenu';
 import { menuBuyer } from '../../../../constants/information';
 import { registrationCommonSlice } from '../../../../store/reducers/registrationCommon';
 import { Route } from '../../../../routers/route';
+import userIcon from '../../../../assets/icons/user-bar/user.svg';
 
 const UserPanel: FC = () => {
```
[the remark this answers is not in the document]
Comment (osaulenkoUA) [reply src/components/shared/Header/UserBar/index.tsx:15]: really why? I don't know, not my code, I will rename it.

File: src/components/shared/Header/UserBar/index.tsx [hunk 8]
```diff
@@ -44,14 +46,14 @@ const UserPanel: FC = () => {
       </Link>
```
[the remark this answers is not in the document]
Comment (osaulenkoUA) [reply src/components/shared/Header/UserBar/index.tsx:46]: For me any errors, all works. It wasn't me implemented

File: src/components/shared/ModalRegistration/AuthForm/index.tsx (hunk 2 above)
Comment (osaulenkoUA) [reply src/components/shared/ModalRegistration/AuthForm/index.tsx:30]: For me any errors, all works.

File: src/store/reducers/userSlice.ts (hunk 6 above)
Comment (osaulenkoUA) [reply src/store/reducers/userSlice.ts:4]: But the user can be defined with any fields

File: src/components/shared/Header/UserBar/index.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (osaulenkoUA) [reply src/components/shared/Header/UserBar/index.tsx:46]: image

**Pull request merged**